extensions/vertx-web/deployment/src/main/java/io/quarkus/vertx/web/deployment/HandlerDescriptor.java

```diff
@@ -31,6 +36,39 @@ boolean isReturningMulti() {
         return method.returnType().name().equals(DotNames.MULTI);
     }
 
+    /**
+     * @return {@code true} if the method is annotated with a constraint or {@code @Valid} or any parameter has such kind of
+     *         annotation.
+     */
+    boolean requireValidation() {
+        if (validationAnnotations == null) {
+            return false;
+        }
+        for (AnnotationInstance annotation : method.annotations()) {
+            String name = annotation.name().toString();
+            if (validationAnnotations.getAllAnnotations().contains(name)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @return {@code true} if the method is annotated with a constraint or {@code @Valid}.
```

Comment:
This is incorrect, you only test for Valid.

Reply:
Fixed